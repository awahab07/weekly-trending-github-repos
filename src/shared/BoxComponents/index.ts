export { default as BoxGrid } from './BoxGrid';
export * from './BoxGrid';
