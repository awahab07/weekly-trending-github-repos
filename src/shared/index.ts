export * from './interfaces/repository';
export * from './utils/Persistence';
