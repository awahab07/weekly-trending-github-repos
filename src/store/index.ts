export { default, acceptHotStore } from './store';
export * from './root.reducer';
