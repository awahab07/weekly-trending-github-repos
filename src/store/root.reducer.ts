import { combineReducers } from '@reduxjs/toolkit';

import { repositoryReducer } from '../features/repository';

const rootReducer = combineReducers({
  repositorySlice: repositoryReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
