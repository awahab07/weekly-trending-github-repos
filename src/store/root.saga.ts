import { all, fork } from 'redux-saga/effects';
import { repositoryListener } from '../features/repository';

export const rootSaga = function* root(): Generator {
  yield all([fork(repositoryListener)]);
};
