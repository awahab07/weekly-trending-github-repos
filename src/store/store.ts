import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import { HmrAcceptor } from '../app/hmr';

import rootReducer from './root.reducer';
import { rootSaga } from './root.saga';

const sagaMiddleware = createSagaMiddleware();

const middleware = [...getDefaultMiddleware({ thunk: false }), sagaMiddleware];

const store = configureStore({
  reducer: rootReducer,
  middleware,
});

sagaMiddleware.run(rootSaga);

export const acceptHotStore = (accept: HmrAcceptor): void => {
  accept('./root.reducer', (): void => {
    const newRootReducer = require('./root.reducer').default;
    store.replaceReducer(newRootReducer);
  });
};

export default store;
