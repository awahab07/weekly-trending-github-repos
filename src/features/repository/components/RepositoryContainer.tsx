import React, { FunctionComponent, PropsWithChildren, ReactElement, useEffect } from 'react';
import { Helmet } from 'react-helmet';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import { IRepository } from '../../../models';
import { RootState } from '../../../store';
import { repositoryActions } from '../index';

const RepositoryContainer: FunctionComponent = (props: PropsWithChildren<{}>): ReactElement => {
  const dispatch = useDispatch();
  const { repositories } = useSelector((state: RootState) => state.repositorySlice, shallowEqual);

  useEffect(() => {
    dispatch(repositoryActions.getRepositories({ filters: [] }));
  }, []);

  const seoTitle = 'Github | Popular Repositories';

  return (
    <>
      <Helmet>{seoTitle}</Helmet>
      {repositories.map((r: IRepository) => (
        <p key={r.id}>{r.name}</p>
      ))}
    </>
  );
};

export default RepositoryContainer;
