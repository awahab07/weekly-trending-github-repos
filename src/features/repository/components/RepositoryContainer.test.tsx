import { mount } from 'enzyme';
import 'jest';
import React from 'react';
import { StaticRouter } from 'react-router';
import { MOCK_REPOS } from '../../../shared/testing';
import { MockStoreProvider } from '../../../store/test';
import RepositoryContainer from './RepositoryContainer';

declare const global: any;

global.window = Object.create(window);
const url = 'http://test.com/category/product';
Object.defineProperty(window, 'location', {
  value: {
    href: url
  }
});

describe('Vip renders correctly', () => {
  it('should render correctly', () => {
    const Component = (
      <MockStoreProvider
        getState={() => ({ repositorySlice: { repositories: MOCK_REPOS as any, loading: false, error: null } })}
      >
        <StaticRouter>
          <RepositoryContainer />
        </StaticRouter>
      </MockStoreProvider>
    );
    const mounted = mount(Component);

    expect(mounted).toMatchSnapshot();
  });
});
