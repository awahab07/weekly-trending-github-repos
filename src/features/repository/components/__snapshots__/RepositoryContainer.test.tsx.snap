// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vip renders correctly should render correctly 1`] = `
<MockStoreProvider
  getState={[Function]}
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <StaticRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        staticContext={Object {}}
      >
        <RepositoryContainer>
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <SideEffect(NullComponent)
              defer={true}
              encodeSpecialCharacters={true}
            >
              <NullComponent
                defer={true}
                encodeSpecialCharacters={true}
              />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <p
            key="1122"
          >
            Test-Repo-Name
          </p>
        </RepositoryContainer>
      </Router>
    </StaticRouter>
  </Provider>
</MockStoreProvider>
`;
